samples/dnn/mask_rcnn.cpp

```diff
@@ -0,0 +1,198 @@
+// Deep learning based Object Detection and Instance Segmentation using Mask R-CNN
+// VIDEO 360 DEMO: https://www.youtube.com/watch?v=0tU8991QgE8
+// Downloads:
+// http://download.tensorflow.org/models/object_detection/mask_rcnn_inception_v2_coco_2018_01_28.tar.gz
+// https://raw.githubusercontent.com/opencv/opencv_extra/master/testdata/dnn/mask_rcnn_inception_v2_coco_2018_01_28.pbtxt
+// https://raw.githubusercontent.com/spmallick/learnopencv/master/Mask-RCNN/mscoco_labels.names
+
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <string>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+
+using namespace cv;
+using namespace dnn;
+using namespace std;
+
+string keys =
+    "{ help  h     | | Print help message. \nUsage \n\t\t./mask_rcnn --image=logo.jpg \n\t\t ./mask_rcnn --video=teste.mp4}"
+    "{ image m     |<none>| Path to input image file.  }"
+    "{ video v     |<none>| Path to input video file.  }"
+    "{ classes n   |./mscoco_labels.names | Path to a text file with names of classes.  }"
+    "{ model w     |./mask_rcnn_inception_v2_coco_2018_01_28/frozen_inference_graph.pb | The pre-trained weights.  }"
+    "{ config c    |./mask_rcnn_inception_v2_coco_2018_01_28.pbtxt | iThe text graph file that has been tuned by the OpenCV’s DNN support group  }"
+    "{ cthr        | .5 | Confidence threshold. }"
+    "{ mthr        | .4 | Mask threshold. }";
+
+void drawBox(Mat& frame, int classId, float conf, Rect box, Mat& objectMask, float maskThreshold, vector<Scalar> colors, vector<string> classes);
+
+void postprocess(Mat& frame, const vector<Mat>& outs, float confThreshold, float maskThreshold, vector<Scalar> colors, vector<string> classes);
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+    parser.about("This sample demonstrates instance segmentation network called Mask-RCNN.");
+    if (argc == 1 || parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    float confThreshold;
+    float maskThreshold;
+    vector<string> classes;
+    vector<Scalar> colors;
+
+
+    confThreshold = parser.get<float>("cthr");
+    maskThreshold = parser.get<float>("mthr");
+
+    String textGraph = parser.get<string>("config");
+    String modelWeights = parser.get<string>("model");
+    string classesFile = parser.get<string>("classes");
+
+    ifstream ifs(classesFile.c_str());
+    string line;
+    while (getline(ifs, line)) classes.push_back(line);
+
+    for(int _c =0; _c<=10; _c++)
+    {
+        colors.push_back( Scalar(rand() % 255,rand() % 255,rand() % 255 ,255.0));
+    }
+
+    Net net = readNet(modelWeights, textGraph,"tensorflow");
```

Comment:
We keep the last argument only for extremal cases when `readNet` cannot determine network origin. In this example it's not necessary due it's convenient to name TensorFlow models with `.pb` extension.

Reply:
ok, done